### PR TITLE
Fix v8.

### PR DESCRIPTION
**Pull request by kobrineli**
1. Patch .gyp files to build targets instead of archiving all the object files into static library.
2. Modify v8 target code to be suitable for libFuzzer and to avoid use-after-free error.

### CODE REVIEW

File: projects/nodejs/build.sh [hunk 1]
```diff
@@ -24,15 +24,9 @@ export CXXFLAGS="-g -std=c++17 -fsanitize=fuzzer-no-link,address,integer,undefin
...
+./configure --with-ossfuzz
 make -j$(nproc)
-ar -rcT static.a $(find . -name "*.o")
-
-export CXXFLAGS="-g -std=c++17 -fsanitize=fuzzer,address,integer,undefined,bounds,null,float-divide-by-zero"
-NODE_WANT_INTERNALS=1 $CXX $CXXFLAGS -pthread test/fuzzers/fuzz_env.cc -o /load_env_fuzzer -DNODE_WANT_INTERNALS \
-     -I./deps/v8/include -I./src/ -I./test/fuzzers/ -I./deps/uv/include/ static.a -ldl -latomic -fno-rtti
-NODE_WANT_INTERNALS=1 $CXX $CXXFLAGS -pthread test/fuzzers/fuzz_url.cc -o /load_url_fuzzer -DNODE_WANT_INTERNALS \
-     -I./deps/v8/include -I./src/ -I./test/fuzzers/ -I./deps/uv/include/ static.a -ldl -latomic -fno-rtti
+cp out/Release/load_env_fuzzer out/Release/load_url_fuzzer out/Release/v8_compile_fuzzer /
```
Comment (SweetVishnya) [inline projects/nodejs/build.sh:29]: cp out/Release/*_fuzzer

Comment (SweetVishnya) [reply projects/nodejs/build.sh:29]: and below

File: projects/nodejs/v8_compile.toml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# limitations under the License.
+#
+################################################################################
+[sydr]
+args = "-l debug"
+target = "/v8_compile_sydr @@"
+
+[libfuzzer]
+path = "/v8_compile_fuzzer"
+args = "-detect_leaks=0 -rss_limit_mb=8192 -jobs=1000 -dict=/js.dict /corpus"
```
Comment (SweetVishnya) [inline projects/nodejs/v8_compile.toml:22]: jobs 1000 without workers?

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: projects/nodejs/build.sh (hunk 1 above)
Comment (kobrineli) [reply projects/nodejs/build.sh]: done

File: projects/nodejs/v8_compile.toml (hunk 2 above)
Comment (kobrineli) [reply projects/nodejs/v8_compile.toml]: done

**Review by SweetVishnya — APPROVED** [review]

**Pull request merged**